module/Api/src/Domain/QueryHandler/Messaging/ApplicationLicenceList/ByOrganisation.php

```diff
@@ -14,26 +16,36 @@
 use Dvsa\Olcs\Transfer\Query\QueryInterface;
 use Dvsa\Olcs\Transfer\Query\Messaging\ApplicationLicenceList\ByOrganisation as ByOrganisationQuery;
 
-class ByOrganisation extends AbstractQueryHandler implements ToggleRequiredInterface
+class ByOrganisation extends AbstractQueryHandler implements ToggleRequiredInterface, AuthAwareInterface
 {
     use ToggleAwareTrait;
+    use AuthAwareTrait;
 
     protected $toggleConfig = [FeatureToggle::MESSAGING];
 
     protected $extraRepos = [Repository\Application::class, Repository\Licence::class];
 
     /**
+     * Fetch a list of licences and applications associated with an organisation.
+     *
+     * If an organisation is not defined in the query, we fall back to the current identity's organisation.
+     *
      * @param ByOrganisationQuery $query
      */
     public function handleQuery(QueryInterface $query): array
     {
         $licenceRepository = $this->getLicenceRepository();
         $applicationRepository = $this->getApplicationRepository();
 
-        $licences = $licenceRepository->fetchByOrganisationId($query->getOrganisation());
+        $orgId = (int)$query->getOrganisation();
+        if (empty($orgId)) {
```

Comment:
C: Compare to 0 if you're forcing the cast above, while empty(0) is true, it reads weird! Maybe just `$query->getOrganisation(); ?: $this->getCurrentOrganisation()->getId()`